src/common/websocket/api.ts

```diff
@@ -71,6 +58,47 @@ export default class DerivWS {
         }
     }
 
+    private messageHandler = (msg: { data: string }) => {
+        const response: DerivAPIResponse = JSON.parse(msg.data)
+        const reqId = response.req_id
+
+        if (reqId && reqId in this.pendingRequests) {
+            this.pendingRequests[reqId].resolve(response)
+            delete this.pendingRequests[reqId]
+        }
+    }
+
+    private recursiveConnect = () => {
```

Comment:
I think this is not an optimal approach for keeping the connection alive, using the `ping/pong` method is less resource consuming, as we know the web socket connection will be close on 2-3 minutes mark, that's when the recursive connect is called. 
the web socket connection is based on `HTTP` and we `upgrade` the connection, so when you open a new connection with `recursiveConnect` there will be at least 2 `HTTP` request, one for initial connection and one for `upgrade` ( handshake ) there will be other requests for handshake validation and etc. please look up `web socket handshake` in google or check out [this article](https://dev.to/hgsgtk/learn-websocket-handshake-protocol-with-gorillawebsocket-server-10k9)

I needed to implement this in one of my tasks, it's not merged yet, but I did it like so: 
https://github.com/binary-com/deriv-com/pull/3679/files#:~:text=private%20ping%20%3D,%2C%201000)

we can talk more on this if you want :pray:

Reply:
Decided to use your approach. Thanks for info and guidance 👍